spec/features/custom_fields/hierarchy_custom_field_spec.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+require "support/pages/custom_fields/hierarchy_page"
+
+RSpec.describe "custom fields of type hierarchy", :js, :with_cuprite do
+  let(:user) { create(:admin) }
+  let(:custom_field_index_page) { Pages::CustomFields::IndexPage.new }
+  let(:new_custom_field_page) { Pages::CustomFields::NewPage.new }
+  let(:hierarchy_page) { Pages::CustomFields::HierarchyPage.new }
+
+  it "lets you create, update and delete a custom field of type hierarchy",
```

Comment:
🔴 Wouldn't be better to split those cases into separate tests?

Reply:
No, IMHO that's exactly the opposite of what we should do with e2e tests.

Reasoning:
The ramp up of the test suite takes ages, compared to the actual test run. This is why I wrote the work with the custom field like a story. You start by creating it, do a lot of nasty stuff to it, and then you end by deleting it. Also, an expensive test like this is meant to test UI and integration, not all of the use cases. Which is why I kept some things out for now.

Reply:
I'd separate them to avoid eventual state leaks to change the behaviour. 

I tend to see feature specs as user stories with clear intent in each test and acceptance criteria for it. We may need to involve others in this discussion.

@apfohl @brunopagno WDYT?

Reply:
In the past I was bitten by having a single/few tests because not all errors would be visible right away. Then you get to the situation where you fix something and it explodes again further into the test.

I understand the argument against, though, that tests might take much longer.

I guess it's a question if we can "pay the price" of longer running tests.

Reply:
I see also two sides. The one side don't wanting an even longer running test suite and the other side wanting to have easier maintainable specs. It seems like you can't have both at the moment. Honestly I would say we should go the way of doing it like @Kharonus. Define the domain story (you could even draw it) and play that thing through. Maybe we could use `# region REGION_NAME` `# endregion` to visually highlight the steps. The IDE allows to collapse them.

In general I would not delay on feature spec development to much. Invest more into automated domain model unit testing.

Reply:
Is separating the test into clearly distinct `regions` something you would approve @mereghost ?

Reply:
I'd say it would alleviate the issue at least. 👍🏼